Copter: Add configurable delay between arming and motor start

jschall opened the pull request.

magicrub commented on ArduCopter/motors.cpp:
```diff
@@ -253,24 +260,53 @@ void Copter::init_disarm_motors()
 // motors_output - send output to motors library which will adjust and send to ESCs and servos
 void Copter::motors_output()
 {
+    static bool old_interlock_status;
```
Best to pull this into the copter class. We try to avoid static variables

rmackay9 commented:
Ok so instead of delaying when the arming happens, we arm but don't let the props spin for one additional second. That makes sense.

R-Lefebvre commented:
Yes.  I also like that it expands the concept of motors spinning is a double-action safety system.  The copter must be armed.  And the motors must be enabled/interlock engaged.  Many people will continue to have the interlock enabled automatically on arming instead of using a switch.  But this is a step in the right direction.

I would like to see the day where the "Safety Switch" would only stop motors from spinning, but not stop servos from moving.  Bring the logic into the vehicle code, instead of the IO code.

In this way, the safety switch could make the vehicle safe by prevent motor start, yet still allow servo movement.  This allows users to ground test servos on airplanes, helicopters and VTOL, as well as multirotors could test Gimbal functions.

jschall commented:
Updated this PR:
- cleaned up the interlock logic in motors_output
- fixed up the timer to properly handle wrapping
- changed it to default to 0 seconds to make it a non-functional change - Randy can decide what he wants for the APM community

jschall commented:
Also modified should_disarm_on_failsafe to return true if we are in the arming delay.

jschall commented:
adding need-rework until this is rebased

jschall commented:
@rmackay9 7 month ping :)

magicrub commented:
PROTIP: looking at this diff with "?w=1" on the URL makes it easier to read :)

jschall commented:
Yes, like this: https://github.com/ArduPilot/ardupilot/pull/3494/files?w=1

proficnc commented:
This is cool.  Is this going in soon?

rmackay9 commented:
Merged, thanks!
sorry for the delay!
